src/v2/cookbook/how-to-use-the-vue-plugin-api.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Making use of the Vue Plugin API
+type: cookbook
+order: 15
+---
+
+## Base Example
+
+Vue exposes a really nifty API for creating plugins, plugins allow us to modify and add features to Vue. This cookbook is aimed at showing you how to take a simple idea and turn it into a full-fledged Vue plugin.
+
+Let's say we want to create a simple abstraction for using Vuex within Vue components, we want to reduce the boilerplate essentially. We can first design how we want the abstraction to look.
+
+Usually we would use Vuex like this in a Vue component:
+
+```js
+import { mapState, mapMutations } from 'vuex'
+
+export default {
+  computed: {
+    ...mapState({
+      user: (state) => state.user,
+    }),
+    ...mapMutations(['UPDATE_USER']),
+  },
+}
+```
+
+Our abstraction would look something like this:
+
+```js
+export default {
+  vuex: {
+    state: {
+      user: (state) => state.user,
+    },
+    mutations: ['UPDATE_USER'],
+  },
+}
+```
+
+## Let's build the plugin
+
+### Setting up the plugin
+
+Before we can get into the logic of the plugin, we first need to setup the plugin in a `.js` file
+
+```js
+import Vue from 'vue'
+
+const plugin = {
+  install(LocalVue) {
```

Comment:
I think it's worth a comment here or sentence above to explain what `LocalVue` is because it's probably difficult for them to infer at this point.